File: udc.m
clc; clear all; close all;
pkg load database;
retry= true;
retry1= true;
retry2= true;
retry3= true;
while retry
  try
    fprintf("\n Elija una opcion \n 1.Correr el programa \n 2.Mostrar Historial \n 3.Salir\n");
    m=(input('Elija:   '));
    if m>3||m==0|m<0
      fprintf("Elija una opcion correcta \n");
    endif
    if m==1
      a1="0";
      retry1=true;
      while retry1
        try
          fprintf("Ingrese un numero entre 1y 999 \n");
          a1=input("Numero:   ","s");
          if a1>999 || a1<=0
            fprintf("\nRevise el valor ingresado\n");
          else
            retry1=false;
          endif
        catch
          fprintf("\n A ingresado un valor o Caracter no valido por favor, ingrese una opcion correcta:\n");
        end_try_catch
      endwhile
      
      c=a1(1);
      d=a1(2);
      u=a1(3);
      fprintf("Valor de centenas: %s \n", c)
      fprintf("Valor de decenas: %s \n" , d)
      fprintf("Valor de unidades: %s \n", u)
      params= cell(1,3);
      uu=str2double(u);
      dd=str2double(d);
      cc=str2double(c);
      params{1,1}= uu;
      params{1,2}= dd;
      params{1,3}= cc;
      conn = pq_connect(setdbopts('dbname','EP1','host','localhost',
      'port','5432','user','postgres','password','123fgthg'));
      N=pq_exec_params(conn, "insert into udc(u, d, c) values($1,$2,$3);",params); %insertar datos en la tabla
      
    endif
    if m==2
      conn = pq_connect(setdbopts('dbname','EP1','host','localhost',
      'port','5432','user','postgres','password','123fgthg'));
      N=pq_exec_params(conn, "Select * from udc;"); %insertar datos en la tabla
      disp(N)
      retry=false;
    endif
    
    if m==3
      retry=false;
    endif
  catch
    fprintf("A ingresado un valor erroneo vuelva a intentarlo %d \n");
    msg = lasterror.message;
    fprintf(msg);
  end_try_catch
endwhile

File: UDC.py
import psycopg2
from tabulate import tabulate
def  post(x,y,z):
    try:
        conexion = psycopg2.connect(
            host = "localhost",
            port = "5432",
            user = "postgres",
            password = "123fgthg",
            dbname = "EP1"
            )
    except:
        print("Sin Conexion Exitosa\n")
            
    cursor = conexion.cursor()
    cursor.execute("INSERT INTO udc(u, d, c) VALUES (%s, %s, %s);", (x, y, z))
    conexion.commit()
    cursor.close()
    conexion.close()

def ret():
    try:
        conexion = psycopg2.connect(
            host = "localhost",
            port = "5432",
            user = "postgres",
            password = "123fgthg",
            dbname = "EP1"
            )
    except:
        print("Sin Conexion Exitosa\n")
    
    cursor = conexion.cursor()
    cursor.execute("SELECT * from udc;")
    print(tabulate(cursor, headers=["ID", "UNIDAD", "DECENA", "CENTENA"], tablefmt="fancy_grid", numalign ="center"))

print("UNIDADES DECENAS Y CENTENAS \n")
cumple = ""
edad=0
while True:
    try:
        print("Elija una opcion:\n1.Correr el Programa\n2.Mostrar el Historial\n3.Salir")
        b=int(input("Elija:\n"))
        if b>3 or b<=0:
            print("Valor ingresado fuera del rango del menu.\n")
        elif b==1:
            while True:
                try:
                    print("Ingrese el numero entre 1 y 999")
                    a1 = int(input("Numero: \n"))
                    if a1>999 or a1<=0:
                        print("Revise el valor ingresado\n")
                        continue
                    break
                except:
                    print("\n A ingresado un valor o Caracter no valido por favor, ingrese una opcion correcta:\n")
                    continue

            c=(a1%1000-a1%100)//100
            d=(a1%100-a1%10)//10
            u=a1%10
            print("Valor de centenas: %d \n" %(c))
            print("Valor de decenas: %d \n" %(d))
            print("Valor de unidades: %d \n"%(u))
            post(u,d,c)
        elif b==2:
            ret()
        elif b==3:
            break

    except Exception as e:
        print("\n A ingresado un valor o Caracter no valido por favor, ingrese una opcion correcta:\n")
        print(repr(e))

exit(0)